src/business_logic/transaction/objects/internal_deploy_account.rs

```diff
@@ -183,6 +189,30 @@ impl InternalDeployAccount {
         }
     }
 
+    fn handle_nonce<S: Default + State + StateReader + Clone>(
+        &self,
+        state: &mut S,
+    ) -> Result<(), TransactionError> {
+        if self.version > 0x8000_0000_0000_0000 {
```

Comment:
This will always be false since self.version is an u64,
Reading the error message I think the if should be version == 0

Reply:
> This will always be false since self.version is an u64,

Not really, since an `u64` can reach `0xFFFF_FFFF_FFFF_FFFF`, which is greater than `0x8000_0000_0000_0000`.

> Reading the error message I think the if should be version == 0

I copied this from `InternalDeclare`, if it's wrong here it'll be wrong there too. Checking with the Python version, that seems to be the case. Fixed.